source/common/http/conn_manager_impl.cc

```diff
@@ -775,15 +777,50 @@ void ConnectionManagerImpl::ActiveStream::decodeData(ActiveStreamDecoderFilter*
 
   for (; entry != decoder_filters_.end(); entry++) {
     ASSERT(!(state_.filter_call_state_ & FilterCallState::DecodeData));
+
+    // We check the request_trailers_ pointer here in case addDecodedTrailers
+    // is called in decodeData - at which point we communicate to the filter
```

Comment:
"is called in decodeData during a previous filter invocation, at which point we communicate to the current and future filters that the stream has not yet ended."